core/platform/email/requests.py

```diff
@@ -0,0 +1,48 @@
+# coding: utf-8
+#
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Provide methods to send HTTP requests (Only POST for now). """
+
+
+import base64
+import urllib
+import urllib2
+
+
+def auth_str(username, password):
+    """Generate authentication string."""
+    if isinstance(username, unicode):
+        username = username.encode('latin1')
+    if isinstance(password, unicode):
+        password = password.encode('latin1')
+
+    return 'Basic ' + base64.b64encode(b':'.join((username, password))).strip()
+
+
+def auth_header(username, password):
+    """Generate authentication header, as a dictionary."""
+    assert isinstance(username, basestring)
+    assert isinstance(password, basestring)
+
+    auth = auth_str(username, password)
+    return {'Authorization': auth}
+
+
+def post(server, auth, data):
```

Comment:
Instead of creating a new module for this, let's just inline a post_to_mailgun() function in mailgun_email_services.py -- no need to pass any args except for `data`. That function can get the mailgun_domain_name, and handle 'api', etc. internally. (That way you can also get rid of a bit of the extra validation above.)

Include, in the function docstring, an explanation that this is adapted from the code for post() in the `requests` library, and provide an Args section (and a Returns section if necessary).

Thanks!

Reply:
Done in commit `Remove requests module and add inline function post_to_mailgun
`